test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt

```diff
@@ -0,0 +1,779 @@
+    fun composedQuery_withDescriptor() {
+        val value1 = 1
+        val value2 = 2
+        val value3 = 3
+        val value4 = 4
+        val value5 = 5
+        val joe = "Joe"
+        val sylvia = "Sylvia"
+        val stacy = "Stacy"
+        val ruth = "Ruth"
+        val bob = "Bob"
+
+        realm.writeBlocking {
+            copyToRealm(
+                Sample().apply {
+                    intField = value1
+                    stringField = joe
+                }
+            )
+            copyToRealm(
+                Sample().apply {
+                    intField = value2
+                    stringField = sylvia
+                }
+            )
+            copyToRealm(
+                Sample().apply {
+                    intField = value3
+                    stringField = stacy
+                }
+            )
+            copyToRealm(
+                Sample().apply {
+                    intField = value4
+                    stringField = ruth
+                }
+            )
+            copyToRealm(
+                Sample().apply {
+                    intField = value5
+                    stringField = bob
+                }
+            )
+        }
+
+        realm.query(Sample::class)
+            .query("intField > 2")
+            .sort(Sample::intField.name, Sort.DESCENDING)
+            .find()
+            .let { results ->
+                assertEquals(3, results.size)
+                assertEquals(value5, results[0].intField)
+                assertEquals(value4, results[1].intField)
+                assertEquals(value3, results[2].intField)
+            }
+
+        realm.query(Sample::class)
+            .query("intField > 2")
+            .sort(Sample::intField.name, Sort.DESCENDING)
+            .limit(1)
+            .find()
+            .let { results ->
+                assertEquals(1, results.size)
+                assertEquals(value5, results[0].intField)
+            }
+
+        // Descriptor in query string - Bob, Ruth, Stacy
+        realm.query(Sample::class)
+            .query("intField > 2 SORT(stringField ASCENDING)")
+            .find()
+            .let { results ->
+                assertEquals(3, results.size)
+                assertEquals(bob, results[0].stringField)
+                assertEquals(ruth, results[1].stringField)
+                assertEquals(stacy, results[2].stringField)
+            }
+
+        // Descriptor as a function - Bob, Ruth, Stacy
+        realm.query(Sample::class)
+            .query("intField > 2")
+            .sort(Sample::stringField.name, Sort.ASCENDING)
+            .find()
+            .let { results ->
+                assertEquals(3, results.size)
+                assertEquals(bob, results[0].stringField)
+                assertEquals(ruth, results[1].stringField)
+                assertEquals(stacy, results[2].stringField)
+            }
+
+        // Conflicting descriptors, query string vs function - Bob, Ruth, Stacy
```

Comment:
Missing case where you are comining string and function descriptors where they both take effect.